File: scala-advent-of-code/src/test/scala/advent/twenty_fifteen/Day14ReindeerOlympicsTest.scala
package advent.twenty_fifteen

import org.scalatest.{FlatSpec, Matchers}

class Day14ReindeerOlympicsTest extends FlatSpec with Matchers {
  "2015 Day 14 - Input" should "calculate answers" in {
    val dp = new Day14ReindeerOlympics("/twenty_fifteen/Day14-ReindeerOlympics-input.txt", 2503)

    dp.part1Answer should be(2640)
    dp.part2Answer should be(1102)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day14ReindeerOlympics("/twenty_fifteen/Day14-ReindeerOlympics-example#1.txt", 1000)

    dp.part1Answer should be(1120)
    dp.part2Answer should be(689)
  }
}

File: java-advent-of-code/src/test/java/advent/twenty_fifteen/Day14ReindeerOlympicsTest.java
package advent.twenty_fifteen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class Day14ReindeerOlympicsTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day14ReindeerOlympics("/twenty_fifteen/Day14-ReindeerOlympics-input.txt", 2503);

        assertEquals(2640, dp.getPart1Answer().intValue());
        assertEquals(1102, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day14ReindeerOlympics("/twenty_fifteen/Day14-ReindeerOlympics-example#1.txt", 1000);

        assertEquals(1120, dp.getPart1Answer().intValue());
        assertEquals(689, dp.getPart2Answer().intValue());
    }
}